src/ripple/overlay/impl/Message.cpp

```diff
@@ -36,23 +39,102 @@ Message::Message (::google::protobuf::Message const& message, int type)
 
     assert (messageBytes != 0);
 
-    /** Number of bytes in a message header. */
-    std::size_t constexpr headerBytes = 6;
+    buffer_.resize (headerBytes + messageBytes);
 
-    mBuffer.resize (headerBytes + messageBytes);
+    setHeader(buffer_.data(), messageBytes, type, Compressed::Off);
 
-    auto ptr = mBuffer.data();
+    if (messageBytes != 0)
+        message.SerializeToArray(buffer_.data() + headerBytes, messageBytes);
+}
 
-    *ptr++ = static_cast<std::uint8_t>((messageBytes >> 24) & 0xFF);
-    *ptr++ = static_cast<std::uint8_t>((messageBytes >> 16) & 0xFF);
-    *ptr++ = static_cast<std::uint8_t>((messageBytes >> 8) & 0xFF);
-    *ptr++ = static_cast<std::uint8_t>(messageBytes & 0xFF);
+void
+Message::compress()
+{
+    std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
C++-17 can infer the template parameter for this. Just `std::lock_guard lock(mutex_);` should work.

Reply:
Done